http/client.go

```diff
@@ -355,6 +356,12 @@ func (c *Client) ExecRequest(
 		return result
 	}
 	err = c.http.setDefaultHeaders(req)
+
+	encConf := encryption.GetContextConfig(ctx)
+	if encConf.HasValue() && encConf.Value().IsEncrypted {
```

Comment:
thought: if the encrypted param only applies to one or two functions it may make more sense as a query parameter instead.

Reply:
do you mean as a parameter in POST request body?

Reply:
No, I think a query param like `?encrypt=true` could be a better option here. I typically use headers for things that apply to a broad set of functions like identity and transactions.